sklearn/preprocessing/imputation.py

```diff
@@ -375,3 +375,202 @@ def transform(self, X):
             X[coordinates] = values
 
         return X
+
+
+class MissingIndicator(BaseEstimator, TransformerMixin):
+    """Binary indicators for missing values.
+
+    Parameters
+    ----------
+    missing_values : integer or np.NaN, optional (default=np.NaN)
+        The placeholder for the missing values. All occurrences of
+        ``missing_values`` will be imputed.
+
+    features : {'train' (default), 'all', array-like of int}
+        If "all", mask will represent all features.
+        If "train", mask will only represent features with missing values
+        during fit time.
+        If mask/indices, mask will only represent features in the
+        indices or mask.
+
+    sparse : boolean or "auto", optional (default="auto")
+        If True, the transformed ``X`` will be a sparse matrix.
+        If False, the transformed ``X`` will be a numpy array.
+        If "auto", the transformed ``X`` will be of same type as input.
+
+    Example
+    -------
+    >>> from sklearn.preprocessing import MissingIndicator
+    >>> import numpy as np
+    >>> X1 = np.array([
+    ...       [np.NaN,  1,  3],
+    ...       [ 4,  0, np.NaN],
+    ...       [ 8,  1,  0]
+    ... ])
+    >>> X2 = np.array([
+    ...       [ 5, np.NaN, np.NaN],
+    ...       [np.NaN,  2,  3],
+    ...       [ 2,  4,  0]
+    ... ])
+    >>> indicator = MissingIndicator(missing_values=np.NaN)
+    >>> indicator.fit(X1)
+    MissingIndicator(features='train', missing_values=nan, sparse='auto')
```

Comment:
Yeah you are right but it just throws a Warning but doesn't raise an error. Should this be changed to throw an error ?